src/day8/re_frame/trace.cljs

```diff
@@ -215,43 +216,61 @@
                                           tag-str))]]
                                [:td.trace--meta.trace--details-icon
                                   {:on-click #(.log js/console tags)}]]))))))))
+
 (defn render-trace-panel []
   (let [filter-input               (r/atom "")
         filter-items               (r/atom (localstorage/get "filter-items" []))
         filter-type                (r/atom :contains)
         input-error                (r/atom false)
+        categories                 (r/atom #{:event :sub/run :sub/create})
         trace-detail-expansions    (r/atom {:show-all? false :overrides {}})]
     (add-watch filter-items
                :update-localstorage
                (fn [_ _ _ new-state]
                  (localstorage/save! "filter-items" new-state)))
     (fn []
-      (let [visible-traces       (if (= @filter-items [])
-                                   @traces
-                                   (filter (apply every-pred (map query->fn @filter-items)) @traces))
+      (let [toggle-category-fn   (fn [category-keys]
+                                   (swap! categories #(set (if (empty? (set/intersection % category-keys))
+                                                             (concat category-keys %)
```

Comment:
Using `concat` and `remove` here will coerce `categories` to a lazy seq. Use `clojure.set` operations if you want to keep the datatype.

Reply:
ahh that's right, thank you! I made the changes to use `clojure.set`opertions instead